src/cli/cli.cpp

```diff
@@ -4891,32 +4884,43 @@ void Interpreter::ProcessLine(char *aBuf)
     VerifyOrExit(argsLength >= 1);
 
 #if OPENTHREAD_CONFIG_DIAG_ENABLE
-    VerifyOrExit((!otDiagIsEnabled(mInstance) || (strcmp(args[0], "diag") == 0)),
+    VerifyOrExit((!otDiagIsEnabled(mInstance) || (args[0] == "diag")),
                  OutputLine("under diagnostics mode, execute 'diag stop' before running any other commands."));
 #endif
 
-    command = Utils::LookupTable::Find(args[0], sCommands);
+    command = Utils::LookupTable::Find(args[0].GetString(), sCommands);
 
     if (command != nullptr)
     {
         OutputResult((this->*command->mHandler)(argsLength - 1, &args[1]));
         ExitNow();
     }
 
-    // Check user defined commands if built-in command has not been found
+    SuccessOrExit(ProcessUserCommands(argsLength, args));
+    OutputResult(OT_ERROR_INVALID_COMMAND);
+
+exit:
+    return;
+}
+
+otError Interpreter::ProcessUserCommands(uint8_t aArgsLength, Arg *aArgs)
```

Comment:
```suggestion
otError Interpreter::ProcessUserCommands(uint8_t aArgsLength, Arg aArgs[])
```
nit: prefer a little bit.

Reply:
I also agree. Makes it clear it's an array. Changed it in new pushed commit.